com.unity.netcode.gameobjects/Runtime/Simulator/INetworkSimulatorScenario.cs

```diff
@@ -1,4 +1,15 @@
-﻿using System.Threading.Tasks;
+﻿// NetSim Implementation compilation boilerplate
+// All references to UNITY_MP_TOOLS_NETSIM_ENABLED should be defined in the same way,
+// as any discrepancies are likely to result in build failures
+// ---------------------------------------------------------------------------------------------------------------------
+#if UNITY_EDITOR || ((DEVELOPMENT_BUILD && !UNITY_MP_TOOLS_NETSIM_DISABLED_IN_DEVELOP) || (!DEVELOPMENT_BUILD && UNITY_MP_TOOLS_NETSIM_ENABLED_IN_RELEASE))
+#define UNITY_MP_TOOLS_NETSIM_ENABLED
```

Comment:
Do we have to do this multiple times because they're in different assemblies or something?

Reply:
Since we can't have a global define, unless explicitly set through scripting symbols, we need this in every file. We also have this all over RNSM files. And since we can't change (AFAIK) scripting symbols based on development or release build, we need this.